File: drl_contracts/src/contracts.rs
pub trait SingleAgentEnv {
    fn state_id(&self) -> usize;
    fn is_game_over(&self) -> bool;
    fn act_with_action_id(&mut self, action_id: usize);
    fn score(&self) -> f32;
    fn available_actions_ids(&self) -> Vec<usize>;
    fn reset(&mut self);
    fn view(&self);
    fn reset_random(&mut self);
}

pub trait MDPEnv {
    fn states(&self) -> Vec<usize>;
    fn actions(&self) -> Vec<usize>;
    fn rewards(&self) -> Vec<f32>;
    fn is_state_terminal(&self, s: usize) -> bool;
    fn transition_probability(&self, s: usize, a: usize, s_p: usize, r: f32) -> f32;
    fn view_state(&self, s: usize);
}

File: drl_sample_project_python/do_not_touch/contracts.py
import numpy as np


class MDPEnv:
    def states(self) -> np.ndarray:
        pass

    def actions(self) -> np.ndarray:
        pass

    def rewards(self) -> np.ndarray:
        pass

    def is_state_terminal(self, s: int) -> bool:
        pass

    def transition_probability(self, s: int, a: int, s_p: int, r: float) -> float:
        pass

    def view_state(self, s: int):
        pass


class SingleAgentEnv:
    def state_id(self) -> int:
        pass

    def is_game_over(self) -> bool:
        pass

    def act_with_action_id(self, action_id: int):
        pass

    def score(self) -> float:
        pass

    def available_actions_ids(self) -> np.ndarray:
        pass

    def reset(self):
        pass

    def view(self):
        pass

    def reset_random(self):
        pass
